functions/module_metadata/assert.pp

```diff
@@ -0,0 +1,69 @@
+# Fails a compile if the client system is not compatible with the module's
+# `metadata.json`
+#
+# @param module_name
+#   The name of the module that should be checked
+#
+# @param options
+#   @see $simplib::module_metadata::blacklist::options
+#
+# @return [None]
+#
+function simplib::module_metadata::assert (
+  String[1] $module_name,
+  Optional[Struct[{
+    enable => Optional[Boolean],
```

Comment:
too many ways to enable/disable.  I would eliminate the global enable, if you really want individual enables

Reply:
@lnemsick-simp The global is for just killing the whole thing across the board. Otherwise you have to keep track of the (potentially updated) API as new capabilities get added.